src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int currentCapacity;
+    private int currentIndex;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void add(T value) {
-
+        addElementToArray(value, currentIndex);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > currentIndex) {
+            throw new ArrayIndexOutOfBoundsException(" Index above the size of Array");
+        }
+        sizeCheck(index);
+        if (index < currentIndex) {
+            System.arraycopy(arrayList, index, arrayList, index + 1, currentIndex + 1);
+        }
+        addElementToArray(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            addElementToArray(list.get(i), currentIndex);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= currentIndex) {
+            throw new ArrayIndexOutOfBoundsException(("No element with index"));
+        }
+        return arrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (currentIndex <= index) {
+            throw new ArrayIndexOutOfBoundsException("Index above Array size");
+        }
+        sizeCheck(index);
+        arrayList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T removedElement = get(index);
+        System.arraycopy(arrayList, index + 1, arrayList, index, currentIndex - 1);
+        currentIndex--;
+        return removedElement;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < currentIndex; i++) {
+            if (t == arrayList[i] || (t != null && t.equals(arrayList[i]))) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("Such element does not exist.");
     }
 
     @Override
     public int size() {
-        return 0;
+        return currentIndex;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return currentIndex == 0;
+    }
+
+    public void addElementToArray(T value, int index) {
```

Comment:
you have to use this method only in one place. So we can keep this logic there

Reply:
Which method? addElementToArray? If it is related to your comment above - I still keep using it in two places when we need to add an element into the middle (or somewhere inside) of the array.